python/amici/sbml_import.py

```diff
@@ -884,6 +937,98 @@ def _process_time(self) -> None:
 
         self._replace_in_all_expressions(sbml_time_symbol, amici_time_symbol)
 
+    def _convert_event_assignment_parameter_targets_to_species(self):
+        """
+        Convert parameters that are targets of event assignments to species.
+
+        This is for the convenience of only implementing event assignments for
+        "species".
+        """
+        parameter_targets = \
+            _collect_event_assignment_parameter_targets(self.sbml)
+        for parameter_target in parameter_targets:
+            # Parameter rate rules already exist as species.
+            if parameter_target in self.symbols[SymbolId.SPECIES]:
+                continue
+            if parameter_target in self.parameter_assignment_rules:
+                raise SBMLException(
+                    'AMICI does not currently support models with SBML events'
+                    'that affect parameters that are also the target of '
+                    'assignment rules.'
+                )
+            parameter_def = None
+            for symbol_id in {SymbolId.PARAMETER, SymbolId.FIXED_PARAMETER}:
+                if parameter_target in self.symbols[symbol_id]:
+                    # `parameter_target` should only exist in one of the
+                    # `symbol_id` dictionaries.
+                    if parameter_def is not None:
+                        raise KeyError(
```

Comment:
```suggestion
                        raise AssertionError(
```
Seems more appropriate.